charts/k8s-monitoring/templates/_config_validations.tpl

```diff
@@ -2,15 +2,54 @@
 {{- if and .Values.logs.enabled .Values.logs.pod_logs.enabled }}
   {{- if eq .Values.logs.pod_logs.gatherMethod "volumes" }}
     {{- if ne (index .Values "alloy-logs").controller.type "daemonset" }}
+{{/*
+Invalid configuration for gathering pod logs! When using logs.pod_logs.gatherMethod: "volumes", Grafana Alloy for Logs must be a Daemonset. Otherwise, logs will be missing!
+Please set:
+logs:
+  pod_logs:
+    gatherMethod: api
+or
+alloy-logs:
+  controller:
+    type: daemonset
+*/}}
       {{ fail "Invalid configuration for gathering pod logs! When using logs.pod_logs.gatherMethod: \"volumes\", Grafana Alloy for Logs must be a Daemonset. Otherwise, logs will be missing!\nPlease set:\nlogs:\n  pod_logs:\n    gatherMethod: api\n  or\nalloy-logs:\n  controller:\n    type: daemonset"}}
     {{- end }}
+    {{- if (index .Values "alloy-logs").alloy.clustering.enabled }}
+{{/*
+Invalid configuration for gathering pod logs! When using logs.pod_logs.gatherMethod: "volumes", Grafana Alloy for Logs should not utilize clustering. Otherwise, performance will suffer!
+Please set:
+alloy-logs:
+  alloy:
+    clustering:
+      enabled: false
+*/}}
+      {{ fail "Invalid configuration for gathering pod logs! When using logs.pod_logs.gatherMethod: \"volumes\", Grafana Alloy for Logs should not utilize clustering. Otherwise, performance will suffer!\nPlease set:\nalloy-logs:\n  alloy:\n    clustering:\n      enabled: false"}}
+    {{- end }}
+
   {{- else if eq .Values.logs.pod_logs.gatherMethod "api" }}
     {{- if not (index .Values "alloy-logs").alloy.clustering.enabled }}
       {{- if eq (index .Values "alloy-logs").controller.type "daemonset" }}
```

Comment:
Logs via k8s API can be collected with a deployment or statefulset, a daemonset ties the number of instances to number of nodes, which is not necessary. Technically this is not wrong though.